futures-util/src/stream/futures_unordered.rs

```diff
@@ -533,13 +535,26 @@ impl<'a, T> Clone for NodeToHandle<'a, T> {
     }
 }
 
+#[doc(hidden)]
+impl<'a, T> From<NodeToHandle<'a, T>> for LocalWaker {
+    fn from(handle: NodeToHandle<'a, T>) -> LocalWaker {
+        unsafe {
+            let ptr: Arc<Node<T>> = handle.0.clone();
+            let ptr: *mut ArcNode<T> = mem::transmute(ptr);
+            let ptr = mem::transmute(ptr as *mut UnsafeWake); // Hide lifetime
+            LocalWaker::new(NonNull::new(ptr).unwrap())
+        }
+    }
+}
+
 #[doc(hidden)]
 impl<'a, T> From<NodeToHandle<'a, T>> for Waker {
     fn from(handle: NodeToHandle<'a, T>) -> Waker {
         unsafe {
-            let ptr = handle.0.clone();
-            let ptr = mem::transmute::<Arc<Node<T>>, *const ArcNode<T>>(ptr);
-            Waker::new(hide_lt(ptr))
+            let ptr: Arc<Node<T>> = handle.0.clone();
+            let ptr: *mut ArcNode<T> = mem::transmute(ptr);
+            let ptr = mem::transmute(ptr as *mut UnsafeWake); // Hide lifetime
```

Comment:
Can you supply manual type arguments to `mem::transmute`? It's hard to tell what it's transmuting between here.